tools/pubsys/src/aws/validate_ssm/mod.rs

```diff
@@ -0,0 +1,732 @@
+//! The validate_ssm module owns the 'validate-ssm' subcommand and controls the process of validating
+//! SSM parameters and AMIs
+
+pub mod results;
+
+use self::results::{SsmValidationResult, SsmValidationResultStatus, SsmValidationResults};
+
+use super::ssm::ssm::get_parameters_by_prefix;
+use super::ssm::{SsmKey, SsmParameters};
+use crate::aws::client::build_client_config;
+use crate::Args;
+use aws_sdk_ssm::{Client as SsmClient, Region};
+use log::{info, trace};
+use pubsys_config::InfraConfig;
+use serde::Deserialize;
+use snafu::ResultExt;
+use std::collections::{HashMap, HashSet};
+use std::fs::File;
+use std::path::PathBuf;
+use structopt::{clap, StructOpt};
+
+/// Validates SSM parameters and AMIs
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+pub struct ValidateSsmArgs {
+    /// File holding the validation configuration
+    #[structopt(long, parse(from_os_str))]
+    validation_config_path: PathBuf,
+
+    // Optional path where the validation results should be written
+    #[structopt(long, parse(from_os_str))]
+    write_results_path: Option<PathBuf>,
+
+    #[structopt(long)]
+    // Optional filter to only write validation results with these statuses to the above path
```

Comment:
Actually important so that structopt will add this information to the help messages.

Also, **list the available options** in this comment.

```suggestion
    /// Optional filter to only write validation results with these statuses to the above path
```